crates/miden-objects/src/account/storage/map.rs

```diff
@@ -171,4 +170,29 @@ mod tests {
         // If these values don't match, update the constants.
         assert_eq!(StorageMap::default().root(), EMPTY_STORAGE_MAP_ROOT);
     }
+
+    #[test]
+    fn account_storage_fail() {
+        // StorageMap for default types (empty map)
+        let storage_map_default = StorageMap::default();
+        let bytes = storage_map_default.to_bytes();
+        assert_eq!(storage_map_default, StorageMap::read_from_bytes(&bytes).unwrap());
+
+        // StorageMap with values
+        let storage_map_leaves_2: [(RpoDigest, Word); 2] = [
+            (
+                RpoDigest::new([Felt::new(101), Felt::new(102), Felt::new(103), Felt::new(104)]),
+                [Felt::new(1_u64), Felt::new(2_u64), Felt::new(3_u64), Felt::new(4_u64)],
+            ),
+            (
+                RpoDigest::new([Felt::new(101), Felt::new(102), Felt::new(103), Felt::new(104)]),
+                [Felt::new(5_u64), Felt::new(6_u64), Felt::new(7_u64), Felt::new(8_u64)],
+            ),
+        ];
+        // let storage_map = StorageMap::with_entries(storage_map_leaves_2);
+
+        let result = catch_unwind(|| StorageMap::with_entries(storage_map_leaves_2));
```

Comment:
Instead of catching the panic, we can assert the error, e.g. something like `assert_matches!(error, MerkleError::DuplicateValuesForIndex(_));`.